cypher/cypher-docs/src/docs/dev/ql/administration/security/security-and-indexes-intro.asciidoc

```diff
@@ -0,0 +1,125 @@
+Ideally, if the labels and properties in the index are denylisted, we can correctly return zero results from both native indexes and full-text indexes.
+However, there are borderline cases where this is not as simple.
+
+Imagine the following nodes were added to the database:
+
+[source, cypher]
+----
+CREATE (:User {name:'Mark', surname:'Andy'});
+CREATE (:User {name:'Andy', surname:'Anderson'});
+CREATE (:User:Person {name:'Mandy', surname:'Smith'});
+CREATE (:User:Person {name:'Joe', surname:'Andy'});
+----
+
+Consider denylisting on the label `:Person`.
+
+[source, cypher]
+----
+DENY TRAVERSE Person ON GRAPH * TO users;
+----
+
+If the user runs a query that will use the native index:
+
+[source, cypher]
+----
+MATCH (n:User) WHERE n.name CONTAINS 'ndy' RETURN n.name;
+----
+
+This query will perform several checks:
+
+* do a scan on the index to create a stream of results of nodes with the `name` property, which leads to four results
+* filter the results to include only nodes where `n.name CONTAINS 'ndy'`, filtering out `Mark` and `Joe` so we have two results
+* filter the results to exclude nodes that also have the label `:Person`, filtering out `Mandy` so we only have one result
+
+For the above dataset, we can see we will get one result.
+
+What if we query this with the full-text index:
+
+[source, cypher]
+----
+CALL db.index.fulltext.queryNodes("userNames", "ndy") YIELD node, score
+RETURN node.name
+----
+
+The problem now is that we do not know if the results provided by the index were because of a match to the `name` or the `surname` property.
+The steps taken by the query engine would be:
+
+* run a _Lucene_ query on the full-text index to produce results containing `ndy` in either property, leading to four results.
+* filter the results to exclude nodes that also have the label `:Person`, filtering out `Mandy` and `Joe` so we have two results.
+
+This difference in results is due to the `OR` relationship between the two properties in the index creation.
+
+==== Blacklisting properties
+
+Now consider denying access on properties, like the `surname` property:
+
+[source, cypher]
+----
+DENY READ {surname} ON GRAPH * TO users;
+----
+
+Now we run the same queries again:
+
+[source, cypher]
+----
+MATCH (n:User) WHERE n.name CONTAINS 'ndy' RETURN n.name;
+----
+
+This query will operate exactly as before, returning the same single result, because nothing in this query relates to the denylisted property.
+
+But consider the full-text index query:
+
+[source, cypher]
+----
+CALL db.index.fulltext.queryNodes("userNames", "ndy") YIELD node, score
+RETURN node.name
+----
+
+The problem now is that we do not know if the results provided by the index were because of a match to the `name` or the `surname` property.
+Results from the surname need to be excluded by the security rules, because they require that the user cannot see any `surname properties.
```

Comment:
```suggestion
Results from the surname need to be excluded by the security rules, because they require that the user cannot see any `surname` properties.
```

Reply:
This still looks wrong when I build